docs/defining_new_slo.md

```diff
@@ -0,0 +1,191 @@
+
+### 1. Choose source of the events
+#### Querying Prometheus
+This mode is mostly just for unification of the SLO computation so everyone uses the same tooling.
+It could be replaced by Prometheus recording rules and just hooking into the alerting by using the same metric naming schema as slo-exporter.
+It also does not support HA setup, so it's not advised to use this and avoid it if possible.
+Slo-exporter documentation can be found [here](https://github.com/seznam/slo-exporter/blob/master/docs/modules/prometheus_ingester.md) and complete example with whole configuration and kubernetes manifests
+[here](https://github.com/seznam/slo-exporter/tree/master/examples/prometheus).
+
+> Warning: in this setup, the slo-exporter cannot run in multiple replicas because it would lead to duplicating the observed events. Make sure to alert on that.
+
+#### From Envoy access logs
+If your service is accessed by users via Envoy proxy, you can configure it to send the access logs using gRPC to the slo-exporter.
+Documentation about the slo-exporter gRPC logging module is [here](https://github.com/seznam/slo-exporter/blob/master/docs/modules/envoy_access_log_server.md),
+documentation how to configure Envoy to send the access logs over gRPC can be found [here](https://www.envoyproxy.io/docs/envoy/latest/api-v3/data/accesslog/v3/accesslog.proto).
+Complete example of the setup is [here](https://github.com/seznam/slo-exporter/tree/master/examples/envoy_proxy).
+
+#### Elasticsearch
+> Currently in experimental mode
+
+In case you want to compute the SLO based on your application logs and use an ELK stack, you can hook the slo-exporter to it.
+
+> Warning: in this setup, the slo-exporter cannot run in multiple replicas because it would lead to duplicating the observed events. Make sure to be alert on that.
+
+#### Tailing file logs
+Similar to those previous but requires the slo-exporter to be running as a sidecar to each observed service instance and tail its application logs.
+slo-exporter documentation [here](https://github.com/seznam/slo-exporter/blob/master/docs/modules/tailer.md)
+and a full example [here](https://github.com/seznam/slo-exporter/tree/master/examples/nginx_proxy).
+
+#### Reading Kafka topic
+In some cases if you want to avoid logging and need high cardinality data you can send each event to a Kafka topic and slo-exporter would read them from there.
+Slo-exporter expects the messages to be in the particular format described [here](https://github.com/seznam/slo-exporter/blob/master/docs/modules/kafka_ingester.md).
+Full example can be found [here](https://github.com/seznam/slo-exporter/tree/master/examples/kafka).
+
+### 2. Configuring and deploying the slo-exporter
+Based on the decisions made, configure the slo-exporter according to [the documentation](https://github.com/seznam/slo-exporter/blob/master/docs/configuration.md)
+and deploy it for example in kubernetes. Example k8s manifests can be found in [../kubernetes](../kubernetes).
+
```

Comment:
```suggestion
Generally, the pipeline should consist of the following stages, in which you need to:
- filter out the unwanted events (e.g. relabel module)
- normalize events' metadata (e.g. eventMetadataRenamer, relabel modules)
- build events' `event_key` (event_key_generator module)
- classify events (in case you want to apply different SLOs) (metadataClassifier, dynamicClassififer, statisticalClassifier modules)
- evaluate each event against given SLI types - availability, latency (slo-event-producer module)
- export events (prometheusExporter module)
```